lint-scan.php

```diff
@@ -455,8 +360,295 @@ function vipgoci_lint_scan_commit(
+		$filename,
+		$temp_file_name,
+		$file_issues_arr_raw
+	);
+
+	vipgoci_log(
+		'Linting issues details',
+		array(
+			'repo_owner'          => $repo_owner,
+			'repo_name'           => $repo_name,
+			'commit_id'           => $commit_id,
+			'filename'            => $filename,
+			'temp_file_name'      => $temp_file_name,
+			'file_issues_arr'     => $file_issues_arr,
+			'file_issues_arr_raw' => $file_issues_arr_raw,
+		),
+		2
+	);
+
+	/* Get rid of the raw version of issues */
+	unset( $file_issues_arr_raw );
+
+
+	return $file_issues_arr;
+
+}
+
+
+/**
+ * This signature is not ready
+ *
+ * This function will be split into different functions that can be moved to
+ * other files (eg git-repo.php for git function calls)
+ *
+ * @param array $prs_implicated
+ * @param array $filter_extensions
+ * @param string $filter_skip_folders
+ * @param array $options
+ * @param array $commit_skipped_files
+ * @param array $commit_issues_stats
+ * @param array $commit_issues_submit
+ *
+ * @return array
+ */
+function vipgoci_gitrepo_scan_modified_files_only( array $prs_implicated, array $filter_extensions = [], string $filter_skip_folders = '', array $options, array &$commit_skipped_files, array &$commit_issues_stats, array &$commit_issues_submit  ): array {
+
+	/**
+	 * Get modified files in each PR that is implicated by the commit
+	 * Iterate it and compare base with head
+	 */
+
+
+	$local_git_repo = $options['local-git-repo'];
+	$prs_modified_files = array();
+
+	/**
+	 * @todo split param arrays and mout the desired strings:
+	 * -- '\'*.php\' '
+	 * -- '\':!tests1/*\' \':!tests/*\''
+	 */
+	$filter_extensions = '-- \'*.php\'';
+//	$filter_skip_folders = '-- \':!tests1/*\'';
+
+	/**
+	 * Save current branch so that we can checkout it at the end of the process
+	 */
+	$cmd     = sprintf(
+		"git -C %s branch --show-current 2>&1",
+		$local_git_repo
+	);
+	$local_git_branch = shell_exec($cmd);
+
+	foreach ($prs_implicated as $pr_number => $pr) {
+		/**
+		 * @todo these attributions are not required
+		 */
+		$pr_base_ref    = $pr->base->ref; // develop main
+		$pr_head_ref    = $pr->head->ref; // branches that contains that commit at the top of the tree, and has open PRs
+
+		/**
+		 * Checkout PR base branch so that git local gets information to compare
+		 */
+		$cmd     = sprintf(
+			"git -C %s checkout %s 2>&1",
+			$local_git_repo,
+			$pr_base_ref
+		);
+		$checkout_base_output = shell_exec( $cmd );
+
+		/**
+		 * Checkout PR branch so that git local gets information to compare
+		 */
+		$cmd     = sprintf(
+			"git -C %s checkout %s 2>&1",
+			$local_git_repo,
+			$pr_head_ref
+		);
+		$checkout_head_output = shell_exec( $cmd );
+
+		/**
+		 * Compares base and head log to get modified files only
+		 * git -C LOCAL_ATH log -M --no-merges --name-only --pretty=format: develop.. -- '*.php' -- . ':!tests1/*' 2>&1
+		 */
+		$cmd     = sprintf(
+			'git  -C %s log -M --oneline --no-merges --name-only --pretty=format: %s.. %s %s 2>&1',
```

Comment:
Yes, this command works efficiently indeed, I've mounted that thinking of that and to include the filters that are required by the application currently.

Please, fix me if I'm mistaken: 

Based on our call and based on this comment (https://github.com/Automattic/vip-go-ci/pull/208#discussion_r707478922):

- As far as I understand, because not all the commits result in a build, to not miss any file modified in between the builds, we would have to compare the branch -- if that's the case, comparing the commits will only raise issues for the files involved with that commit. 

If we are only to compare the commit that triggered the build with the main last commit, we can use the ``vipgoci_git_diffs_fetch`` function.

Reply:
https://github.com/Automattic/vip-go-ci/pull/212#discussion_r719152672

Reply:
> If we are only to compare the commit that triggered the build with the main last commit, we can use the vipgoci_git_diffs_fetch function.

That should work. This function can though generate a diff between _any_ two commits.

PHPCS scanning uses [this function currently](https://github.com/Automattic/vip-go-ci/blob/main/phpcs-scan.php#L459).